make_travis_yml_2.hs

```diff
@@ -71,29 +73,34 @@ versionNumbers (Version vn _) = vn
 
 #endif
 
-putStrLnErr :: MonadIO m => String -> m a
-putStrLnErr  m = liftIO $ hPutStrLn stderr ("*ERROR* " ++ m) >> exitFailure
+type YamlWriter m a = ReaderT (String -> m ()) (WriterT [String] m) a
```

Comment:
Rather have a tuple to write to, for yaml and diagnostics. This is ugly.

Reply:
This is part of PR #100, which I just updated to match your sense of aesthetics ;)